### DISCUSSION TITLE
Apply cross vhost policy&limit through rabbitmqctl or definition import

### DISCUSSION THREAD

**Discussion by al3xsun** [rabbitmq/rabbitmq-server#4012] [Other] [answered]
Hi, there:

Upon looking into applying a cross-vhost policy or vhost limit (apply to multiple vhosts & automatically gets added when new vhost created), I found this [issue](https://github.com/rabbitmq/rabbitmq-cli/issues/281) with a similar ask. 

I checked out the definition import suggested in the thread and found that the definition import still requires me specifying each vhost name, if I were to specify * for vhost name and import, it would state `Virtual host '*' does not exist`. Please let me know if I am doing this incorrectly

Therefore I have two questions:
1. Is there or would there be support for cross-vhost policy (such that with pattern matching, if vhost gets added after the policy is applied and it matches the vhost name pattern, the policy would get applied to it) 

2. If answer to question 1 is no. Is there a way for me to import a definition that applies certain policy to current and future vhosts added, meaning defining a global policy that limit on all current & future vhosts and import it on broker start?

Thanks a lot for taking a look!

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
Please use GitHub discussions for questions. Issues are for specific, actionable items that core team engineers can work on (when and if we choose to).

**Comment by michaelklishin** [comment, marked as answer]
There is no such thing as a cross-vhost policy. Everything in a virtual host is isolated from the rest. Definition files with policies are JSON documents, and thus easily generated.

**Reply by al3xsun** [reply]
Thanks for the response!

“”Everything in a virtual host is isolated from the rest.“”
Want to confirm that this isolation is a tenet rabbitmq team tries to keep. 

If I want to apply a policy such as max-length for all vhosts (created not all at the same time), the best approach is to set the limit after creating one right? Since there are no ways of applying policy or limit to a vhost that's not created yet through rabbitmq set_policy or set_vhost_limit
